### ISSUE TITLE
logictest, logictestccl, execbuilder tests fail for `geospatial` test-data on ppc64le

### ISSUE THREAD

**Issue by amitsadaphule** [cockroachdb/cockroach#72225]
**Describe the problem**

I built the cockroachdb **v20.2.17** on ppc64le and when I executed the test suite, I found that there were test failures for the following packages:

```
github.com/cockroachdb/cockroach/pkg/ccl/logictestccl
github.com/cockroachdb/cockroach/pkg/sql/logictest
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder
```

When I analysed the logs [geospatial-testdata-failures.zip](https://github.com/cockroachdb/cockroach/files/7441613/geospatial-testdata-failures.zip), I found that the failing tests were related to the `geospatial` test data. A sample log of test failure for `logictest` package pasted below (Please refer full package test logs in attached zip for more details):

```
            --- FAIL: TestLogic/local/geospatial/geom_operators (0.25s)
                logic.go:2130: 
                     
                    testdata/logic_test/geospatial:1991: SELECT
                      a.dsc,
                      b.dsc,
                      ST_HausdorffDistance(a.geom, b.geom),
                      ST_HausdorffDistance(a.geom, b.geom, 0.4)
                    FROM geom_operators_test a
                    JOIN geom_operators_test b ON (1=1)
                    ORDER BY a.dsc, b.dsc
                    expected:
                        Empty GeometryCollection                  Empty GeometryCollection                  NULL               NULL
                        Empty GeometryCollection                  Empty LineString                          NULL               NULL
                        Empty GeometryCollection                  Empty Point                               NULL               NULL
                        Empty GeometryCollection                  Faraway point                             NULL               NULL
                        Empty GeometryCollection                  Line going through left and right square  NULL               NULL
                        Empty GeometryCollection                  NULL                                      NULL               NULL
                        Empty GeometryCollection                  Nested Geometry Collection                NULL               NULL
                        Empty GeometryCollection                  Point middle of Left Square               NULL               NULL
                        Empty GeometryCollection                  Point middle of Right Square              NULL               NULL
                        Empty GeometryCollection                  Square (left)                             NULL               NULL
                        Empty GeometryCollection                  Square (right)                            NULL               NULL
                        Empty GeometryCollection                  Square overlapping left and right square  NULL               NULL
                        Empty LineString                          Empty GeometryCollection                  NULL               NULL
                        Empty LineString                          Empty LineString                          NULL               NULL
                        Empty LineString                          Empty Point                               NULL               NULL
                        Empty LineString                          Faraway point                             NULL               NULL
                        Empty LineString                          Line going through left and right square  NULL               NULL
                        Empty LineString                          NULL                                      NULL               NULL
                        Empty LineString                          Nested Geometry Collection                NULL               NULL
                        Empty LineString                          Point middle of Left Square               NULL               NULL
                        Empty LineString                          Point middle of Right Square              NULL               NULL
                        Empty LineString                          Square (left)                             NULL               NULL
                        Empty LineString                          Square (right)                            NULL               NULL
                        Empty LineString                          Square overlapping left and right square  NULL               NULL
                        Empty Point                               Empty GeometryCollection                  NULL               NULL
                        Empty Point                               Empty LineString                          NULL               NULL
                        Empty Point                               Empty Point                               NULL               NULL
                        Empty Point                               Faraway point                             NULL               NULL
                        Empty Point                               Line going through left and right square  NULL               NULL
                        Empty Point                               NULL                                      NULL               NULL
                        Empty Point                               Nested Geometry Collection                NULL               NULL
                        Empty Point                               Point middle of Left Square               NULL               NULL
                        Empty Point                               Point middle of Right Square              NULL               NULL
                        Empty Point                               Square (left)                             NULL               NULL
                        Empty Point                               Square (right)                            NULL               NULL
                        Empty Point                               Square overlapping left and right square  NULL               NULL
                        Faraway point                             Empty GeometryCollection                  NULL               NULL
                        Faraway point                             Empty LineString                          NULL               NULL
                        Faraway point                             Empty Point                               NULL               NULL
                        Faraway point                             Faraway point                             0                  0
                        Faraway point                             Line going through left and right square  7.10633520177595   7.10633520177595
                        Faraway point                             NULL                                      NULL               NULL
                        Faraway point                             Nested Geometry Collection                7.07106781186548   7.07106781186548
                        Faraway point                             Point middle of Left Square               7.10633520177595   7.10633520177595
                        Faraway point                             Point middle of Right Square              6.36396103067893   6.36396103067893
                        Faraway point                             Square (left)                             7.81024967590665   7.81024967590665
                        Faraway point                             Square (right)                            7.07106781186548   7.07106781186548
                        Faraway point                             Square overlapping left and right square  7.14212853426764   7.14212853426764
                        Line going through left and right square  Empty GeometryCollection                  NULL               NULL
                        Line going through left and right square  Empty LineString                          NULL               NULL
                        Line going through left and right square  Empty Point                               NULL               NULL
                        Line going through left and right square  Faraway point                             7.10633520177595   7.10633520177595
                        Line going through left and right square  Line going through left and right square  0                  0
                        Line going through left and right square  NULL                                      NULL               NULL
                        Line going through left and right square  Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Line going through left and right square  Point middle of Left Square               1                  1
                        Line going through left and right square  Point middle of Right Square              1                  1
                        Line going through left and right square  Square (left)                             0.707106781186548  0.707106781186548
                        Line going through left and right square  Square (right)                            0.707106781186548  0.707106781186548
                        Line going through left and right square  Square overlapping left and right square  0.707106781186548  0.707106781186548
                        NULL                                      Empty GeometryCollection                  NULL               NULL
                        NULL                                      Empty LineString                          NULL               NULL
                        NULL                                      Empty Point                               NULL               NULL
                        NULL                                      Faraway point                             NULL               NULL
                        NULL                                      Line going through left and right square  NULL               NULL
                        NULL                                      NULL                                      NULL               NULL
                        NULL                                      Nested Geometry Collection                NULL               NULL
                        NULL                                      Point middle of Left Square               NULL               NULL
                        NULL                                      Point middle of Right Square              NULL               NULL
                        NULL                                      Square (left)                             NULL               NULL
                        NULL                                      Square (right)                            NULL               NULL
                        NULL                                      Square overlapping left and right square  NULL               NULL
                        Nested Geometry Collection                Empty GeometryCollection                  NULL               NULL
                        Nested Geometry Collection                Empty LineString                          NULL               NULL
                        Nested Geometry Collection                Empty Point                               NULL               NULL
                        Nested Geometry Collection                Faraway point                             7.07106781186548   7.07106781186548
                        Nested Geometry Collection                Line going through left and right square  0.707106781186548  0.707106781186548
                        Nested Geometry Collection                NULL                                      NULL               NULL
                        Nested Geometry Collection                Nested Geometry Collection                0                  0
                        Nested Geometry Collection                Point middle of Left Square               0.707106781186548  0.707106781186548
                        Nested Geometry Collection                Point middle of Right Square              0.707106781186548  0.707106781186548
                        Nested Geometry Collection                Square (left)                             1.4142135623731    1.4142135623731
                        Nested Geometry Collection                Square (right)                            1.4142135623731    1.4142135623731
                        Nested Geometry Collection                Square overlapping left and right square  1.4142135623731    1.4142135623731
                        Point middle of Left Square               Empty GeometryCollection                  NULL               NULL
                        Point middle of Left Square               Empty LineString                          NULL               NULL
                        Point middle of Left Square               Empty Point                               NULL               NULL
                        Point middle of Left Square               Faraway point                             7.10633520177595   7.10633520177595
                        Point middle of Left Square               Line going through left and right square  1                  1
                        Point middle of Left Square               NULL                                      NULL               NULL
                        Point middle of Left Square               Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Point middle of Left Square               Point middle of Left Square               0                  0
                        Point middle of Left Square               Point middle of Right Square              1                  1
                        Point middle of Left Square               Square (left)                             0.707106781186548  0.707106781186548
                        Point middle of Left Square               Square (right)                            1.58113883008419   1.58113883008419
                        Point middle of Left Square               Square overlapping left and right square  1.58113883008419   1.58113883008419
                        Point middle of Right Square              Empty GeometryCollection                  NULL               NULL
                        Point middle of Right Square              Empty LineString                          NULL               NULL
                        Point middle of Right Square              Empty Point                               NULL               NULL
                        Point middle of Right Square              Faraway point                             6.36396103067893   6.36396103067893
                        Point middle of Right Square              Line going through left and right square  1                  1
                        Point middle of Right Square              NULL                                      NULL               NULL
                        Point middle of Right Square              Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Point middle of Right Square              Point middle of Left Square               1                  1
                        Point middle of Right Square              Point middle of Right Square              0                  0
                        Point middle of Right Square              Square (left)                             1.58113883008419   1.58113883008419
                        Point middle of Right Square              Square (right)                            0.707106781186548  0.707106781186548
                        Point middle of Right Square              Square overlapping left and right square  0.781024967590665  0.781024967590665
                        Square (left)                             Empty GeometryCollection                  NULL               NULL
                        Square (left)                             Empty LineString                          NULL               NULL
                        Square (left)                             Empty Point                               NULL               NULL
                        Square (left)                             Faraway point                             7.81024967590665   7.81024967590665
                        Square (left)                             Line going through left and right square  0.707106781186548  0.707106781186548
                        Square (left)                             NULL                                      NULL               NULL
                        Square (left)                             Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square (left)                             Point middle of Left Square               0.707106781186548  0.707106781186548
                        Square (left)                             Point middle of Right Square              1.58113883008419   1.58113883008419
                        Square (left)                             Square (left)                             0                  5.55111512312578e-17
                        Square (left)                             Square (right)                            1                  1
                        Square (left)                             Square overlapping left and right square  1                  1
                        Square (right)                            Empty GeometryCollection                  NULL               NULL
                        Square (right)                            Empty LineString                          NULL               NULL
                        Square (right)                            Empty Point                               NULL               NULL
                        Square (right)                            Faraway point                             7.07106781186548   7.07106781186548
                        Square (right)                            Line going through left and right square  0.707106781186548  0.707106781186548
                        Square (right)                            NULL                                      NULL               NULL
                        Square (right)                            Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square (right)                            Point middle of Left Square               1.58113883008419   1.58113883008419
                        Square (right)                            Point middle of Right Square              0.707106781186548  0.707106781186548
                        Square (right)                            Square (left)                             1                  1
                        Square (right)                            Square (right)                            0                  5.55111512312578e-17
                        Square (right)                            Square overlapping left and right square  0.1                0.1
                        Square overlapping left and right square  Empty GeometryCollection                  NULL               NULL
                        Square overlapping left and right square  Empty LineString                          NULL               NULL
                        Square overlapping left and right square  Empty Point                               NULL               NULL
                        Square overlapping left and right square  Faraway point                             7.14212853426764   7.14212853426764
                        Square overlapping left and right square  Line going through left and right square  0.707106781186548  0.707106781186548
                        Square overlapping left and right square  NULL                                      NULL               NULL
                        Square overlapping left and right square  Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square overlapping left and right square  Point middle of Left Square               1.58113883008419   1.58113883008419
                        Square overlapping left and right square  Point middle of Right Square              0.781024967590665  0.781024967590665
                        Square overlapping left and right square  Square (left)                             1                  1
                        Square overlapping left and right square  Square (right)                            0.1                0.1
                        Square overlapping left and right square  Square overlapping left and right square  0                  5.55111512312578e-17
                        
                    but found (query options: "") :
                        Empty GeometryCollection                  Empty GeometryCollection                  NULL               NULL
                        Empty GeometryCollection                  Empty LineString                          NULL               NULL
                        Empty GeometryCollection                  Empty Point                               NULL               NULL
                        Empty GeometryCollection                  Faraway point                             NULL               NULL
                        Empty GeometryCollection                  Line going through left and right square  NULL               NULL
                        Empty GeometryCollection                  NULL                                      NULL               NULL
                        Empty GeometryCollection                  Nested Geometry Collection                NULL               NULL
                        Empty GeometryCollection                  Point middle of Left Square               NULL               NULL
                        Empty GeometryCollection                  Point middle of Right Square              NULL               NULL
                        Empty GeometryCollection                  Square (left)                             NULL               NULL
                        Empty GeometryCollection                  Square (right)                            NULL               NULL
                        Empty GeometryCollection                  Square overlapping left and right square  NULL               NULL
                        Empty LineString                          Empty GeometryCollection                  NULL               NULL
                        Empty LineString                          Empty LineString                          NULL               NULL
                        Empty LineString                          Empty Point                               NULL               NULL
                        Empty LineString                          Faraway point                             NULL               NULL
                        Empty LineString                          Line going through left and right square  NULL               NULL
                        Empty LineString                          NULL                                      NULL               NULL
                        Empty LineString                          Nested Geometry Collection                NULL               NULL
                        Empty LineString                          Point middle of Left Square               NULL               NULL
                        Empty LineString                          Point middle of Right Square              NULL               NULL
                        Empty LineString                          Square (left)                             NULL               NULL
                        Empty LineString                          Square (right)                            NULL               NULL
                        Empty LineString                          Square overlapping left and right square  NULL               NULL
                        Empty Point                               Empty GeometryCollection                  NULL               NULL
                        Empty Point                               Empty LineString                          NULL               NULL
                        Empty Point                               Empty Point                               NULL               NULL
                        Empty Point                               Faraway point                             NULL               NULL
                        Empty Point                               Line going through left and right square  NULL               NULL
                        Empty Point                               NULL                                      NULL               NULL
                        Empty Point                               Nested Geometry Collection                NULL               NULL
                        Empty Point                               Point middle of Left Square               NULL               NULL
                        Empty Point                               Point middle of Right Square              NULL               NULL
                        Empty Point                               Square (left)                             NULL               NULL
                        Empty Point                               Square (right)                            NULL               NULL
                        Empty Point                               Square overlapping left and right square  NULL               NULL
                        Faraway point                             Empty GeometryCollection                  NULL               NULL
                        Faraway point                             Empty LineString                          NULL               NULL
                        Faraway point                             Empty Point                               NULL               NULL
                        Faraway point                             Faraway point                             0                  0
                        Faraway point                             Line going through left and right square  7.10633520177595   7.10633520177595
                        Faraway point                             NULL                                      NULL               NULL
                        Faraway point                             Nested Geometry Collection                7.07106781186548   7.07106781186548
                        Faraway point                             Point middle of Left Square               7.10633520177595   7.10633520177595
                        Faraway point                             Point middle of Right Square              6.36396103067893   6.36396103067893
                        Faraway point                             Square (left)                             7.81024967590665   7.81024967590665
                        Faraway point                             Square (right)                            7.07106781186548   7.07106781186548
                        Faraway point                             Square overlapping left and right square  7.14212853426764   7.14212853426764
                        Line going through left and right square  Empty GeometryCollection                  NULL               NULL
                        Line going through left and right square  Empty LineString                          NULL               NULL
                        Line going through left and right square  Empty Point                               NULL               NULL
                        Line going through left and right square  Faraway point                             7.10633520177595   7.10633520177595
                        Line going through left and right square  Line going through left and right square  0                  0
                        Line going through left and right square  NULL                                      NULL               NULL
                        Line going through left and right square  Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Line going through left and right square  Point middle of Left Square               1                  1
                        Line going through left and right square  Point middle of Right Square              1                  1
                        Line going through left and right square  Square (left)                             0.707106781186548  0.707106781186548
                        Line going through left and right square  Square (right)                            0.707106781186548  0.707106781186548
                        Line going through left and right square  Square overlapping left and right square  0.707106781186548  0.707106781186548
                        NULL                                      Empty GeometryCollection                  NULL               NULL
                        NULL                                      Empty LineString                          NULL               NULL
                        NULL                                      Empty Point                               NULL               NULL
                        NULL                                      Faraway point                             NULL               NULL
                        NULL                                      Line going through left and right square  NULL               NULL
                        NULL                                      NULL                                      NULL               NULL
                        NULL                                      Nested Geometry Collection                NULL               NULL
                        NULL                                      Point middle of Left Square               NULL               NULL
                        NULL                                      Point middle of Right Square              NULL               NULL
                        NULL                                      Square (left)                             NULL               NULL
                        NULL                                      Square (right)                            NULL               NULL
                        NULL                                      Square overlapping left and right square  NULL               NULL
                        Nested Geometry Collection                Empty GeometryCollection                  NULL               NULL
                        Nested Geometry Collection                Empty LineString                          NULL               NULL
                        Nested Geometry Collection                Empty Point                               NULL               NULL
                        Nested Geometry Collection                Faraway point                             7.07106781186548   7.07106781186548
                        Nested Geometry Collection                Line going through left and right square  0.707106781186548  0.707106781186548
                        Nested Geometry Collection                NULL                                      NULL               NULL
                        Nested Geometry Collection                Nested Geometry Collection                0                  0
                        Nested Geometry Collection                Point middle of Left Square               0.707106781186548  0.707106781186548
                        Nested Geometry Collection                Point middle of Right Square              0.707106781186548  0.707106781186548
                        Nested Geometry Collection                Square (left)                             1.4142135623731    1.4142135623731
                        Nested Geometry Collection                Square (right)                            1.4142135623731    1.4142135623731
                        Nested Geometry Collection                Square overlapping left and right square  1.4142135623731    1.4142135623731
                        Point middle of Left Square               Empty GeometryCollection                  NULL               NULL
                        Point middle of Left Square               Empty LineString                          NULL               NULL
                        Point middle of Left Square               Empty Point                               NULL               NULL
                        Point middle of Left Square               Faraway point                             7.10633520177595   7.10633520177595
                        Point middle of Left Square               Line going through left and right square  1                  1
                        Point middle of Left Square               NULL                                      NULL               NULL
                        Point middle of Left Square               Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Point middle of Left Square               Point middle of Left Square               0                  0
                        Point middle of Left Square               Point middle of Right Square              1                  1
                        Point middle of Left Square               Square (left)                             0.707106781186548  0.707106781186548
                        Point middle of Left Square               Square (right)                            1.58113883008419   1.58113883008419
                        Point middle of Left Square               Square overlapping left and right square  1.58113883008419   1.58113883008419
                        Point middle of Right Square              Empty GeometryCollection                  NULL               NULL
                        Point middle of Right Square              Empty LineString                          NULL               NULL
                        Point middle of Right Square              Empty Point                               NULL               NULL
                        Point middle of Right Square              Faraway point                             6.36396103067893   6.36396103067893
                        Point middle of Right Square              Line going through left and right square  1                  1
                        Point middle of Right Square              NULL                                      NULL               NULL
                        Point middle of Right Square              Nested Geometry Collection                0.707106781186548  0.707106781186548
                        Point middle of Right Square              Point middle of Left Square               1                  1
                        Point middle of Right Square              Point middle of Right Square              0                  0
                        Point middle of Right Square              Square (left)                             1.58113883008419   1.58113883008419
                        Point middle of Right Square              Square (right)                            0.707106781186548  0.707106781186548
                        Point middle of Right Square              Square overlapping left and right square  0.781024967590665  0.781024967590665
                        Square (left)                             Empty GeometryCollection                  NULL               NULL
                        Square (left)                             Empty LineString                          NULL               NULL
                        Square (left)                             Empty Point                               NULL               NULL
                        Square (left)                             Faraway point                             7.81024967590665   7.81024967590665
                        Square (left)                             Line going through left and right square  0.707106781186548  0.707106781186548
                        Square (left)                             NULL                                      NULL               NULL
                        Square (left)                             Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square (left)                             Point middle of Left Square               0.707106781186548  0.707106781186548
                        Square (left)                             Point middle of Right Square              1.58113883008419   1.58113883008419
                        Square (left)                             Square (left)                             0                  5.55111512312578e-17
                        Square (left)                             Square (right)                            1                  1
                        Square (left)                             Square overlapping left and right square  1                  1
                        Square (right)                            Empty GeometryCollection                  NULL               NULL
                        Square (right)                            Empty LineString                          NULL               NULL
                        Square (right)                            Empty Point                               NULL               NULL
                        Square (right)                            Faraway point                             7.07106781186548   7.07106781186548
                        Square (right)                            Line going through left and right square  0.707106781186548  0.707106781186548
                        Square (right)                            NULL                                      NULL               NULL
                        Square (right)                            Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square (right)                            Point middle of Left Square               1.58113883008419   1.58113883008419
                        Square (right)                            Point middle of Right Square              0.707106781186548  0.707106781186548
                        Square (right)                            Square (left)                             1                  1
                        Square (right)                            Square (right)                            0                  5.55111512312578e-17
                        Square (right)                            Square overlapping left and right square  0.1                0.1
                        Square overlapping left and right square  Empty GeometryCollection                  NULL               NULL
                        Square overlapping left and right square  Empty LineString                          NULL               NULL
                        Square overlapping left and right square  Empty Point                               NULL               NULL
                        Square overlapping left and right square  Faraway point                             7.14212853426764   7.14212853426764
                        Square overlapping left and right square  Line going through left and right square  0.707106781186548  0.707106781186548
                        Square overlapping left and right square  NULL                                      NULL               NULL
                        Square overlapping left and right square  Nested Geometry Collection                1.4142135623731    1.4142135623731
                        Square overlapping left and right square  Point middle of Left Square               1.58113883008419   1.58113883008419
                        Square overlapping left and right square  Point middle of Right Square              0.781024967590665  0.781024967590665
                        Square overlapping left and right square  Square (left)                             1                  1
                        Square overlapping left and right square  Square (right)                            0.1                0.1
                        Square overlapping left and right square  Square overlapping left and right square  0                  1.11022302462516e-16
                logic.go:1685: 
                     testdata/logic_test/geospatial:2146: too many errors encountered, skipping the rest of the input
```

Any pointers on how to resolve this issue will be great help.

**To Reproduce**

Here are the steps to reproduce the issue:

```
#!/bin/bash

CWD=`pwd`

# Install all dependencies
dnf -y --disableplugin=subscription-manager install \
        http://mirror.centos.org/centos/8/BaseOS/ppc64le/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm \
        http://mirror.centos.org/centos/8/BaseOS/ppc64le/os/Packages/centos-linux-repos-8-2.el8.noarch.rpm \
        https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm

yum install -y git cmake make gcc-c++ autoconf ncurses-devel.ppc64le wget.ppc64le openssl-devel.ppc64le diffutils procps-ng

cd $HOME

set -eu

# Install nodejs
NODE_VERSION=v12.18.2
DISTRO=linux-ppc64le
wget https://nodejs.org/dist/$NODE_VERSION/node-$NODE_VERSION-$DISTRO.tar.gz
tar -xzf node-$NODE_VERSION-$DISTRO.tar.gz
export PATH=$HOME/node-$NODE_VERSION-$DISTRO/bin:$PATH

npm install yarn --global

cd $HOME

# Setup go environment and install go
GOPATH=$HOME/go
COCKROACH_HOME=$GOPATH/src/github.com/cockroachdb
mkdir -p $COCKROACH_HOME
export GOPATH
curl -O https://dl.google.com/go/go1.13.5.linux-ppc64le.tar.gz
tar -C /usr/local -xzf go1.13.5.linux-ppc64le.tar.gz
rm -rf go1.13.5.linux-ppc64le.tar.gz
export PATH=$PATH:/usr/local/go/bin

# Clone cockroach and build
COCKROACH_VERSION=v20.2.17
cd $COCKROACH_HOME
git clone https://github.com/cockroachdb/cockroach.git
cd cockroach
git checkout $COCKROACH_VERSION
make buildoss

export GOMAXPROCS=4
make test PKG=./pkg/sql/logictest TESTFLAGS='-v -count=1'
make test PKG=./pkg/ccl/logictestccl TESTFLAGS='-v -count=1'
make test PKG=./pkg/sql/opt/exec/execbuilder TESTFLAGS='-v -count=1'
```

**Environment:**

CockroachDB [v20.2.17]
Architecture and OS: [ppc64le/RHEL/UBI 8.4]
Go: [1.13.5]
Node: [12.18.2]

Any pointers to resolve the failures would be of great help.

Jira issue: CRDB-11047

**Comment by knz** [comment]
@amitsadaphule this is the result of a different floating-point behavior on PPC.

As explained in #72227, CockroachDB's team does not support alternative build platforms, so we're likely not going to work actively to resolve this particular problem. However, on the other hand we have many users that report that cockroachDB works on their platform, even if not officially supported.

Note that the difference in behavior wrt floating-point computations means that SQL queries are likely to return slightly differnt results than on other platforms. However, from experience the difference is often very small and depending on the application might be negligible.

**Comment by knz** [comment]
@amitsadaphule also, in the future please do not file common issues for multiple go packages. We would prefer different issues filed for different areas. Thank you.

**Comment by RaduBerinde** [comment]
CC @otan

**Comment by otan** [comment]
ack, @knz has the right explanation. we've tried to "fix" some of these in PRs such as #66621, but we decided against it.

**Comment by prashantkhoje** [comment]
With following patch execbuilder geospatial test failures seen on ppc64le are resolved:
```
root@p006vm cockroach]# cd /root/go/src/github.com/cockroachdb/cockroach/c-deps/geos
[root@p006vm geos]# git diff
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 1f27b802..2e501b51 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -21,7 +21,12 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/geos-config
         PERMISSIONS
         OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 
-add_subdirectory(astyle)
+option(BUILD_ASTYLE "Build astyle (Artistic Style) tool" OFF)
+
+message(STATUS "GEOS: Build astyle ${BUILD_ASTYLE}")
+if(BUILD_ASTYLE)
+  add_subdirectory(astyle)
+endif()
 
 # if(UNIX OR MINGW)
 
[root@p006vm geos]#
```
References: https://github.com/libgeos/geos/issues/460#issuecomment-870709386, https://github.com/libgeos/geos/issues/461
